geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaInputFormat.scala

```diff
@@ -152,16 +171,18 @@ class GeoMesaInputFormat extends InputFormat[Text, SimpleFeature] {
    *
    * Our delegated AccumuloInputFormat creates a split for each range - because we set a lot of ranges in
    * geomesa, that creates too many mappers. Instead, we try to group the ranges by tservers. We use the
-   * number of shards in the schema as a proxy for number of tservers. We also know that each range will
-   * only have a single location, because we always set autoAdjustRanges in the configure method above.
+   * number of shards in the schema as a proxy for number of tservers.
    */
   override def getSplits(context: JobContext): java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    // try to create 2 mappers per node - account for case where there are less splits than shards
-    val groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2))
+    // fallback on creating 2 mappers per node if desiredSplits is unset.
+    // Account for case where there are less splits than shards
+    var groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2))
+    if (desiredSplitCount != -1) {
+      groupSize = Math.max(1, accumuloSplits.length / desiredSplitCount)
+    }
```

Comment:
another tidbit for the future: you can avoid using a var (generally discouraged in scala) by something like:
val groupSize = if (desiredSplitCount ...) foo else bar